satpy/readers/lsasaf_h5.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Nowcasting SAF common PPS&MSG NetCDF/CF format reader.
+
+References:
+   - The LSASAF products documentation: https://landsaf.ipma.pt/en/data/products/
+
+"""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+from pyproj import CRS
+
+from satpy.readers._geos_area import get_area_definition, get_area_extent
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.utils import get_legacy_chunk_size
+
+logger = logging.getLogger(__name__)
+
+CHUNK_SIZE = get_legacy_chunk_size()
+
+PLATFORM_NAMES = {'MSG1': 'Meteosat-8',
+                  'MSG2': 'Meteosat-9',
+                  'MSG3': 'Meteosat-10',
+                  'MSG4': 'Meteosat-11',
+                  }
+
+INST_NAMES = {'SEVI': 'SEVIRI'}
+
+
+class H5LSASAF(BaseFileHandler):
+    """LSA-SAF MSG HDF5 reader."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init method."""
+        super(H5LSASAF, self).__init__(filename, filename_info,
+                                       filetype_info)
+
+        self.cache = {}
+        self.nc = xr.open_dataset(self.filename,
+                                  decode_cf=True,
+                                  mask_and_scale=False,
+                                  chunks=CHUNK_SIZE)
+
+        self.nc = self.nc.rename({'phony_dim_0': 'x', 'phony_dim_1': 'y'})
+
+        self.platform_name = self.nc.attrs['SATELLITE'][0].rstrip()
+        self.sensor = self._get_inst_name()
+
+    def _get_inst_name(self):
+        """Get the instrument name."""
+        inst_id = self.nc.attrs['INSTRUMENT_ID'][0].rstrip()
+        try:
+            return INST_NAMES[inst_id]
+        except KeyError:
+            raise KeyError(f'Unknown instrument ID: {inst_id}')
+
+    def get_dataset(self, dsid, info):
+        """Load a dataset."""
+        dsid_name = dsid['name']
+        if dsid_name in self.cache:
+            logger.debug('Get the data set from cache: %s.', dsid_name)
+            return self.cache[dsid_name]
```

Comment:
1. Is the cache ever actually written to?
2. If it was, would it really provide a performance improvement?
3. In what cases is the same file handler asked for the same DataArray multiple times?